src/react-tag-list.js

```diff
@@ -54,18 +70,20 @@ var TagList = React.createClass({
     if(this.getCurrentTagHeight() !== this.state.currentTagHeight)
       this.setState({currentTagHeight: this.getCurrentTagHeight()});
 
-    //if the shown count updated, update the state
-    if(this.state.shownCount !== this.getShownCount())
-      this.setState({shownCount: this.getShownCount()});
+    //if the shown count updated, update the state, this should happen when any tags get added
+    if(this.state.shownCount !== this.numberOfTagsShownInCollapsedContainer())
+      this.setState({shownCount: this.numberOfTagsShownInCollapsedContainer()});
```

Comment:
minor gripe: can you save the value of the number of tags shown so you don't have to re-call it?